docs/guides/building-forms.md

```diff
@@ -0,0 +1,387 @@
+This was a quick recap of setting up web forms in HTML.
+
+```html
+<form action="/action_page.php">
+  <label for="pswrd">Password:</label>
+  <input
+    type="password"
+    id="pswrd"
+    name="pswrd"
+    pattern="[a-z]{0,9}"
+    title="Password should be digits (0 to 9) or alphabets (a to z)."
+  />
+
+  <button type="submit">Submit</button>
+</form>
+```
+
+The above example defines how to use the `pattern` attribute. In this case, the password form field must only contain digits (0 to 9) and lowercase alphabets (a to z). No other special characters (#,$,&, etc.) The rule in RegEx is written as `[a-z]{1,15}`.
+
+![form-validate-regex](https://assets.vercel.com/image/upload/c_scale,w_675/v1643009088/nextjs/guides/building-forms/form-validate-regex.jpg)
+
+This was a quick recap of setting up web forms in HTML.
+
+> To learn more about HTML forms, check out the [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Learn/Forms).
+
+## Part 2: [SERVER] Serverless Forms with Next.js and Vercel
+
+A serverless architecture, in literal terms, does involve servers. The difference is that they are managed and hosted by the cloud computing companies like [Amazon Web Services](https://aws.amazon.com/lambda/), [Microsoft Azure](https://docs.microsoft.com/en-us/azure/azure-functions/functions-overview), [Vercel](https://vercel.com/docs/concepts/functions/introduction#serverless-functions), and [Google Cloud Platform](https://cloud.google.com/functions/). Being relatively new to the league, Vercel offers top-notch cloud computing services — the first among them is its ease of use. You can write functions both as a part of its [Functions](https://vercel.com/docs/concepts/functions/introduction) framework and Next.js Framework via [API routes](https://nextjs.org/docs/api-routes/introduction).
+
+Vercel empowers you to write JavaScript Serverless functions and deploy them at its edge network. This significantly minimizes latency because your web application runs code in a serverless architecture as close to the end-user as possible. It works incredibly well if you want to build lightweight and flexible applications that can be expanded or updated quickly.
+
+Having said that, let's jump on to the next part, where you'll be recreating these HTML forms in React using Next.js.
+
+Create a simple Next.js app. You can use the [create-next-app](https://nextjs.org/docs/getting-started#setup) for a quick start. In your command line terminal, run the following:
+
+```
+npx create-next-app
+```
+
+Answer the questions to create your project, and give it a name, this example uses [`next-forms`](https://github.com/vercel/next.js/tree/canary/examples/next-forms). Next `cd` into this directory, and run `npm run dev` or `yarn dev` command to start the development server.
+
+Open the URL printed in the terminal to ensure that your app is running successfully.
+
+## Setting up the Serverless Environment
```

Comment:
Isnt this a part of `Part 2: [SERVER] Serverless Forms with Next.js and Vercel` ? If so we should make this an H3, and correct the following headings within this section of the guide

Reply:
It's basically a separate heading for how to create a Serverless Environment with Next.js and Vercel. It kind of explains generic info about Vercel's serverless functions and Next.js API endpoints. And because it discusses the server-side of things that's why I've added it here.